api/v1beta1/azuremachine_default.go

```diff
@@ -143,15 +173,54 @@ func (s *AzureMachineSpec) SetNetworkInterfacesDefaults() {
 	}
 }
 
+// GetSubscriptionID returns the subscription ID for the given cluster name and namespace.
+func GetSubscriptionID(cli client.Client, clusterName string, namespace string, maxAttempts int) (string, error) {
```

Comment:
should we rename this to withRetry and move it to `util/azure` so it's consistent?

Reply:
I tried moving this to `util/azure`, but that causes an import cycle due to the function needing to import `AzureCluster` from `v1beta1`. I'm not sure if there's a way around that with Go.